### PR TITLE
cisco: qos-sai multi-asic: Add static route to send traffic for multi-asic and multi-dut tests

### PR DESCRIPTION
**Pull request by rraghav-cisco**
These changes are needed to make the multi-asic test variants to work over cisco-8000 platforms.

Cisco-8000 distributed platform uses internal iBGP routing to determine forwarding path for packets destined to other ASICs(same linecard or different linecards) via the Fabric Card. Since the qos-sai setup removes BGP, this routing information is lost. We need to setup a static route in the source-asic so that the packets can be forwarded to the destination ASIC.

When the qos sai script runs, with just a single asic, we can see the routes are all removed from that asic:
root@sfd-vt2-lc0:/home/cisco# ip netns exec asic0 show ip route -n asic0
Codes: K - kernel route, C - connected, S - static, R - RIP,
O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
F - PBR, f - OpenFabric,
> - selected route, * - FIB route, q - queued route, r - rejected route

C>*1.0.0.0/16 is directly connected, eth1, 04:44:46
C>*3.3.3.3/32 is directly connected, Loopback4096, 04:44:42
C>*10.0.0.0/31 is directly connected, PortChannel102, 04:43:33
C>*10.0.0.4/31 is directly connected, PortChannel104, 04:43:33
C>*10.0.0.8/31 is directly connected, PortChannel106, 04:43:33
C>*10.0.0.12/31 is directly connected, PortChannel108, 04:43:33
C>*10.0.0.16/31 is directly connected, PortChannel1010, 04:43:33
C>*10.0.0.20/31 is directly connected, PortChannel1012, 04:43:33
C>*10.1.0.1/32 is directly connected, Loopback0, 04:44:42
root@sfd-vt2-lc0:/home/cisco#

In this PR, we add a static route in the source ASIC to the dst_port_ip via a backplane interface IP address as gateway. This backplane interface IP is detected from the IP address list in the dst ASIC.

### CODE REVIEW

**Comment by rraghav-cisco** [comment]
@abdosi , @rbpittman : FYI.

File: tests/qos/test_qos_sai.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def testParameter(
     def testQosSaiPfcXoffLimit(
         self, xoffProfile, duthosts, get_src_dst_asic_and_duts,
         ptfhost, dutTestParams, dutConfig, dutQosConfig,
-        ingressLosslessProfile, egressLosslessProfile
+        ingressLosslessProfile, egressLosslessProfile, set_static_route
```
[the remark this answers is not in the document]
Comment (rraghav-cisco) [reply tests/qos/test_qos_sai.py:181]: @abdosi : In the beginning only this test was multi-asic aware, but now the fixture itself is autouse=True. So the fixture is going to be used for all the testcases. I have to remove this diff in the testcase.

[the remark this answers is not in the document]
Comment (abdosi) [reply tests/qos/test_qos_sai.py:181]: please remove the diff before we can merge this.

[the remark this answers is not in the document]
Comment (abdosi) [reply tests/qos/test_qos_sai.py:181]: also add the check if topo is t2 and switch_type as chassis-packet we do this static route creation. Do not check for asic -type

File: tests/qos/qos_sai_base.py [hunk 2]
```diff
@@ -1835,3 +1835,80 @@ def dualtor_ports_for_duts(request, get_src_dst_asic_and_duts):
...
         return dualtor_ports_set
+
+    @pytest.fixture(scope='function', autouse=True)
+    def set_static_route(
+            self, get_src_dst_asic_and_duts, dutTestParams, dutConfig):
+        # Get portchannels.
+        # find the one that is backplane based.
+        # set a static route through that portchannel.
+        # remove when done.
+        if dutTestParams["basicParams"]["sonic_asic_type"] != "cisco-8000":
```
Comment (abdosi) [inline tests/qos/qos_sai_base.py:1846]: check for switch_type as chassis-packet and topo as t2. Not asic type. this will make change more generic,

**Review by abdosi — CHANGES_REQUESTED** [review]
as comments

**Review by abdosi — CHANGES_REQUESTED** [review]
as comments

File: tests/qos/qos_sai_base.py (hunk 2 above)
Comment (rraghav-cisco) [reply tests/qos/qos_sai_base.py:1846]: @abdosi , if the switch_type is not provided in the ansible/lab file, should this code be executed?

Comment (rraghav-cisco) [reply tests/qos/qos_sai_base.py]: @abdosi: Fixed, but need to know what to do if the switch_type is not available.

**Review by abdosi — APPROVED** [review]

**Comment by XuChen-MSFT** [comment]
@rraghav-cisco @abdosi , Can you help to cherry pick to master branch and 202305 branch?

+ @wangxin @StormLiangMS  for vis

**Pull request merged**